CorsixTH/Lua/app.lua

```diff
@@ -1680,60 +1680,127 @@ function App:loadLuaFolder(dir, no_results, append_to)
+  bug before the next milestone. The bug is then patched for the next milestone.
+  Moving forward, revisions should have a savegame_version bump of 1, as 10
+  savegame versions are reserved each release for patching.
+  All beta versions must have a savegame increment each time
+  Each patch note must begin with a '-'
+--]]
+local version_table = {
+  -- Format: major, minor, revision, patch (string), savegame_version
+  {major = 0, minor = 0, revision = 8, patch = "", version = 0}, -- Beta 8 or below
+  {major = 0, minor = 1, revision = 0, patch = "", version = 51},
+  {major = 0, minor = 10, revision = 0, patch = "", version = 53},
+  {major = 0, minor = 11, revision = 0, patch = "", version = 54},
+  {major = 0, minor = 20, revision = 0, patch = "", version = 66},
+  {major = 0, minor = 21, revision = 0, patch = "", version = 72},
+  {major = 0, minor = 30, revision = 0, patch = "", version = 78},
+  {major = 0, minor = 40, revision = 0, patch = "", version = 91},
+  {major = 0, minor = 50, revision = 0, patch = "", version = 105},
+  {major = 0, minor = 60, revision = 0, patch = "", version = 111},
+  {major = 0, minor = 61, revision = 0, patch = "", version = 122},
+  {major = 0, minor = 62, revision = 0, patch = "", version = 127},
+  {major = 0, minor = 63, revision = 0, patch = "", version = 134},
+  {major = 0, minor = 64, revision = 0, patch = "", version = 138},
+  {major = 0, minor = 65, revision = 0, patch = "", version = 156},
+  -- There was also 0.65.1, not differentiated by version number
+  {major = 0, minor = 66, revision = 0, patch = "", version = 170},
+  {major = 0, minor = 67, revision = 0, patch = "", version = 180},
+  {major = 0, minor = 68, revision = 0, patch = "", version = 194},
+  {major = 0, minor = 69, revision = 0, patch = "-beta-1", version = 216}
+}
+
+--! Retrieve the current savegame version as defined in the application.
+function App:getCurrentVersion()
+  return self.savegame_version
+end
+
+--! Requests data regarding a given savegame version
+--!param savegame_version (number) What to lookup, uses application version if blank
+--!param concatenate (boolean) (optional) Whether the version should be converted to a string
+--!return the matching version as a table, or as a string if concatenate is true.
+-- If no versions match it returns the base version with the savegame version
+-- attached e.g. 0.66.0-dev172
+function App:getVersion(savegame_version, concatenate)
+  savegame_version = savegame_version or self:getCurrentVersion()
+  local result
+  for i = #version_table, 1, -1 do
+    local version = version_table[i]
+    if version.version == savegame_version then
+      result = version
+      break
+    elseif version.version - savegame_version < 0 then
+      -- we're not on a release version
+      local devel_version = shallow_clone(version) -- prevent recursion
+      devel_version.patch = version.patch .. "-dev" .. savegame_version
+      result = devel_version
```

Comment:
The result is slightly wrong. It would be better to also overwrite `devel_version.version` with the requested version number.